command/hcl2_upgrade.go

```diff
@@ -158,8 +158,8 @@ func (c *HCL2UpgradeCommand) RunContext(_ context.Context, cla *HCL2UpgradeArgs)
 		return 1
 	}
 
-	core := hdl.(*CoreWrapper).Core
```

Comment:
Long story short: can't remember exactly the reason why.
As a rationale though: CoreWrapper's existence is only justified by the fact that legacy JSON templates (defined by the `packer.Core` type) don't return a `hcl.Diagnostics` when they're failing. I think it maybe has outlived its purpose? We can probably simplify the code by removing that extra layer, and I think that's what I went for with this PR.

If you think it still has relevance though I can try rolling that change back and see if we can live with it!

Reply:
Note: I just remembered also, since `CoreWrapper` was in the `command` package, and since `hcp` is imported by `command`, we had a dependency loop. Hence why it was dropped in the first place too